### PR TITLE
chore(changelog): adding changelog for 21.12 and adding tagname

### PR DESCRIPTION
**Pull request by aman-bansal**
---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/8096)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,478 @@ All notable changes to this project will be documented in this file.
...
+  - Feat(GRAPHQL): adds `@default` directive for setting default field values at create and update ([#8017][])
+  - Feat(GRAPHQL): Support auth with custom DQL ([#7775][])
+  - Feat(GRAPHQL): This PR allows updatable and nullable `@id` fields. ([#7736][])
+  - Feat(GRAPHQL): Disallow DQL schema changes for predicates used in GraphQL schema (DGRAPH-3245) ([#7742][])
+  - Feat(GRAPHQL): This PR allows `@id` field in interface to be unique across all the implementing types. ([#7710][])
+  - Feat(GRAPHQL): Add language tag support in GraphQL ([#7663][])
+  - Feat(GRAPHQL): Zero HTTP endpoints are now available at GraphQL admin (GRAPHQL-1118) ([#6649][])
+  - Feat(GRAPHQL): Webhooks on add/update/delete mutations (GRAPHQL-1045) ([#7494][])
+  - Feat(GRAPHQL): Allow Multipe JWKUrls for auth. ([#7528][])
+  - Feat(GRAPHQL): Add support for passing OAuth Bearer token as authorization JWT ([#7490)
```
Comment (danielmai) [inline CHANGELOG.md:36]: Missing link for PR 7490.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,478 @@ All notable changes to this project will be documented in this file.
...
+
+- Core Dgraph
+  - Feat(metrics): Add Badger metrics. ([#8034][])
+  - Feat(magicNumber): Introduce magic number ([#8032][])
+  - Feat(lambda): allow access to access jwt in lambda ([#8023][])
+  - Feat(rdf-response): Support RDF response via http query request ([#8004][])
+  - Feat(sroar): Use rank() API from sroar and some cleanup ([#8002][])
+  - Feat(lambda): store lambda scripts within the dgraph, send it to JS s… ([#7955][])
```
Comment (danielmai) [inline CHANGELOG.md:44]: This entry looks incomplete.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,478 @@ All notable changes to this project will be documented in this file.
...
+  - Feat(backup): Separate out writers from handlers
+  - Feat(Backup): Refactor the way backup code is structured.
+  - Feat(query): Add mechanism to have a limit on number of pending queries ([#7603][])
+  - Feat(flag): remove unused badger.max-retries option from bulk command ([#7591][])
+  - Feat(sentry): clusterID flag added for alpha sentry reports (gql-services) ([#7580][])
+  - Feat(increment tool): Support queries to Dgraph Cloud
+  - Feat(cmd/debuginfo) add new metrics to be collected ([#7439][])
+  - Feat(flags): use Vault for ACL secrets ([#7492][])
+  - Feat(Apollo): Add support for `@provides` and `@requires` directive.  ([#7503][])
+  - Feat(restore): Introduce incremental restore ([#7942) ([#7971][])
```
Comment (danielmai) [inline CHANGELOG.md:63]: Missing link for PR 7942.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,478 @@ All notable changes to this project will be documented in this file.
...
+- Enterprise Features
+  - Feat(Multi-tenancy): Add namespaces field to state ([#7808][])
+  - Feat(multi-tenancy): make drop data namespace aware ([#7789) ([#7795][])
+  - Feat(cdc): Add support for SCRAM SASL mechanism ([#7765][])
+  - Feat(acl): allow access to all the predicates using wildcard ([#7991][])
+
+### Fixed
+
+- GraphQL
+  - Fix(GRAPHQL): add validation of null values with correct order of graphql rule validation ([#8007) ([#8008][])
```
Comment (danielmai) [inline CHANGELOG.md:75]: Missing link for PR 8007

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,478 @@ All notable changes to this project will be documented in this file.
...
+  - Fix(GRAPHQL): fix type assertion failure in graphql if resolver is not defined ([#8003][])
+  - Fix(GRAPHQL): fixing graphql schema update when the data is restored + skippin… ([#7970][])
```
Comment (danielmai) [inline CHANGELOG.md:77]: This entry looks incomplete.

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,478 @@ All notable changes to this project will be documented in this file.
...
+  - Fix(badger): Upgrade badger to fix deadlock ([#8025][])
+  - Fix(bulk): quote the schema correctly in bulk loader ([#8019][])
+  - Fix(sbs): handle response error gracefully ([#8018][])
+  - Fix(admin): make config changes to pass through gog middlewares ([#8014][])
+  - Fix(lambda): fix race condition in lambda server spin up ([#8013][])
+  - Fix(proposals): Incremental proposal key for zero proposals ([#8005][])
+  - Fix(lambda): monitor lambda server, fix performance issue, remove lambda logs from extensions ([#8006][])
+  - Fix(live): quote the xid when doing upsert ([#7983][])
+  - Fix(sroar): Bring latest sroar to master ([#7977][])
+  - Fix(query): Do not execute filters if there are no source uids ([#7962) ([#7969][])
```
Comment (danielmai) [inline CHANGELOG.md:133]: Missing PR link for PR 7962.

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,478 @@ All notable changes to this project will be documented in this file.
...
+  - Fix(snapshot): update last snapshot time across members ([#7968][])
+  - Fix(pool): use write lock when getting health info ([#7963][])
+  - Fix(JoinCluster): Avoid retrying JoinCluster indefinitely ([#7961][])
+  - Fix(rollups): Write rolled-up keys at ts+1 ([#7957) ([#7959][])
+  - Fix(conn): JoinCluster loop should use latest conn ([#7950][])
+  - Fix(restore): Set kv version to restoreTs for rolled up keys and schema keys ([#7930) ([#7935][])
```
Comment (danielmai) [inline CHANGELOG.md:139]: Missing PR link for PR 7930.

File: CHANGELOG.md [hunk 8]
```diff
@@ -4,6 +4,478 @@ All notable changes to this project will be documented in this file.
...
+  - Fix(lambda): fix race condition in lambda server spin up ([#8013][])
+  - Fix(proposals): Incremental proposal key for zero proposals ([#8005][])
+  - Fix(lambda): monitor lambda server, fix performance issue, remove lambda logs from extensions ([#8006][])
+  - Fix(live): quote the xid when doing upsert ([#7983][])
+  - Fix(sroar): Bring latest sroar to master ([#7977][])
+  - Fix(query): Do not execute filters if there are no source uids ([#7962) ([#7969][])
+  - Fix(snapshot): update last snapshot time across members ([#7968][])
+  - Fix(pool): use write lock when getting health info ([#7963][])
+  - Fix(JoinCluster): Avoid retrying JoinCluster indefinitely ([#7961][])
+  - Fix(rollups): Write rolled-up keys at ts+1 ([#7957) ([#7959][])
```
Comment (danielmai) [inline CHANGELOG.md:137]: Missing PR link for PR 7957.

File: CHANGELOG.md [hunk 9]
```diff
@@ -4,6 +4,478 @@ All notable changes to this project will be documented in this file.
...
+  - Fix(restore): Set kv version to restoreTs for rolled up keys and schema keys ([#7930) ([#7935][])
+  - Fix(backup): Fix full backup request ([#7932][])
+  - Fix(cmd/debug): Print banned namespaces correctly. ([#7929][])
+  - Reconnect via a redial in case of disconnection. ([#7918][])
+  - Fix(Raft): Detect network partition when streaming
+  - Fix(metrics): Expose dgraph_num_backups_failed_total metric view. ([#7900][])
+  - Fix(sroar): Fix TestAuthWithCustomDQL failure because of roaring bitmaps ([#7902][])
+  - Fix(DQL): revert changes related to cascade pagination with sort ([#7885][])
+  - Fix(restore): append galaxy namespace to type name ([#7880][])
+  - Fix(Backup): use validReadTs from manifest for backward compatibility ([#7601) ([#7863][])
```
Comment (danielmai) [inline CHANGELOG.md:148]: Missing PR link for PR 7601.

File: CHANGELOG.md [hunk 10]
```diff
@@ -4,6 +4,478 @@ All notable changes to this project will be documented in this file.
...
+  - Fix(restore): consider the banned namespaces while bumping ([#7839][])
+  - Fix(restore): update the schema and type from 2103 ([#7838][])
+  - Fix(updatemanifest): update the version of manifest after update ([#7828][])
+  - Fix(backup): handle manifest version logic, update manifest version to 2105 ([#7825][])
+  - Fix(schema-update): Start opIndexing only when index creation is required. ([#7845][])
+  - Fix(admin): remove exportedFiles field ([#7835][])
+  - Fix(restore): reset the kv.StreamId before sending to stream writer ([#7833][])
+  - Fix(auth): preserve the status code while returning error ([#7832][])
+  - bug fix to permit audit streaming to stdout writer([#7803][])
+  - Fix(lease): don't do rate limiting when not limit is not specified ([#7787) ([#7801][])
```
Comment (danielmai) [inline CHANGELOG.md:159]: Missing PR link for PR 7787.

File: CHANGELOG.md [hunk 11]
```diff
@@ -4,6 +4,478 @@ All notable changes to this project will be documented in this file.
...
+- Opt(schema): Optimize populateSchema() by avoiding repeated lock acquisition ([#8068][])
+- Perf: Speed up parsing of a huge query with a lot of conditional mutations ([#7871][])
+- Opt(Restore): Make restore map phase faster ([#8038][])
+- Opt(codec): return nil instead of a new bitmap ([#7997][])
+- Opt(cache): Use Ristretto to store posting lists ([#7995][])
+- Opt(rdf-output): Make RDF output generation concurrent ([#7988][])
+- Opt(recurse): Optimise recurse and bring range iterators from sroar ([#7989][])
+- Opt(restore): Sort the buffer before spinning the writeToDisk goroutine ([#7984][])
+- Perf(sroar): Use latest sroar and add histogram in the sbs tool ([#7982][])
+- Opt(Alpha): Load schema and types using Stream framework ([#7938) ([#7940][])
```
Comment (danielmai) [inline CHANGELOG.md:245]: Missing PR link for PR 7938.

**Review by danielmai — CHANGES_REQUESTED** [review]
Is https://github.com/dgraph-io/dgraph/pull/8092 part of the changelog? I didn't see it.

**Review by ahsanbarkati — APPROVED** [review]

**Review by danielmai — APPROVED** [review]

**Pull request merged**